tracing/trace_state.hh

```diff
@@ -510,6 +511,24 @@ class opentelemetry_state final {
 private:
     lw_shared_ptr<trace_state> _state_ptr;
     bool const _opentelemetry_tracing{false};
+    inet_address_vector_replica_set _replicas;
+
+    void serialize_replicas(bytes& serialized) const noexcept {
```

Comment:
Please move the implementation to `trace_state.cc`

Reply:
This function should not be marked `noexcept` as appending to `bytes` allocates memory which can fail and throw exception.

Reply:
Similar thing with `noexcept` in `set_replicas()`, although that one can be mitigated by accepting a copy and doing a move-assignment (which is `noexcept` for `small_vector`)